drivers/comparator/comparator_mchp_ac_g1.c

```diff
@@ -0,0 +1,596 @@
+	LOG_DBG("===================================================");
+}
+#endif /* CONFIG_COMPARATOR_LOG_LEVEL_DBG */
+
+/* Wait for synchronization */
+static void ac_wait_sync(ac_registers_t *ac_reg, uint32_t mask)
+{
+	if (WAIT_FOR(((ac_reg->AC_SYNCBUSY & mask) != mask), TIMEOUT_VALUE_US,
+		     k_busy_wait(DELAY_US)) == false) {
+		LOG_ERR("Timeout waiting for AC_SYNCBUSY bits to clear (mask=0x%X)", mask);
+	}
+}
+
+/* AC utility functions for enabling, configuring, and syncing the analog comparator */
+static void ac_enable(ac_registers_t *ac_reg, bool enable)
+{
+	if (enable == true) {
+		ac_reg->AC_CTRLA |= AC_CTRLA_ENABLE_Msk;
+	} else {
+		ac_reg->AC_CTRLA &= ~AC_CTRLA_ENABLE_Msk;
+	}
+
+	ac_wait_sync(ac_reg, AC_SYNCBUSY_ENABLE_Msk);
+}
+
+/* Enable a specific comparator channel and wait for sync */
+static void ac_channel_enable(ac_registers_t *ac_reg, uint8_t channel_id)
+{
+	ac_reg->AC_COMPCTRL[channel_id] |= AC_COMPCTRL_ENABLE_Msk;
+
+	/* Only 2 possible values for channel_id: 0 or 1 */
+	if (channel_id == 0) {
+		ac_wait_sync(ac_reg, AC_SYNCBUSY_COMPCTRL0_Msk);
+	} else {
+		ac_wait_sync(ac_reg, AC_SYNCBUSY_COMPCTRL1_Msk);
+	}
+}
+
+/* Enable interrupt for the given comparator channel */
+static inline void ac_enable_interrupt(ac_registers_t *ac_reg, uint8_t channel_id)
+{
+	if (channel_id == 0) {
+		ac_reg->AC_INTENSET = AC_INTENSET_COMP0_Msk;
+	} else {
+		ac_reg->AC_INTENSET = AC_INTENSET_COMP1_Msk;
+	}
+}
+
+/* Disable interrupt for the given comparator channel */
+static inline void ac_disable_interrupt(ac_registers_t *ac_reg, uint8_t channel_id)
+{
+	if (channel_id == 0) {
+		ac_reg->AC_INTENCLR = AC_INTENCLR_COMP0_Msk;
+	} else {
+		ac_reg->AC_INTENCLR = AC_INTENCLR_COMP1_Msk;
+	}
+}
+
+/* Trigger a single-shot comparison for the specified channel */
+static inline void ac_start_conversion(ac_registers_t *ac_reg, uint8_t channel_id)
+{
+	ac_reg->AC_CTRLB |= (AC_CTRLB_START0_Msk << channel_id);
+}
+
+/* Wait until the comparator result for the specified channel is ready */
+static int ac_wait_for_conversion(ac_registers_t *ac_reg, uint8_t channel_id)
+{
+	uint32_t ready_mask = AC_STATUSB_READY0_Msk << channel_id;
+
+	if (WAIT_FOR(((ac_reg->AC_STATUSB & ready_mask) == ready_mask), TIMEOUT_VALUE_US,
+		     k_busy_wait(DELAY_US)) == false) {
+		LOG_ERR("Timeout waiting for AC_STATUSB ready (channel=%u)", channel_id);
+		return -ETIMEDOUT;
+	}
+
+	return 0;
+}
+
+/* Get the current comparator output state for the specified channel
+ * Returns true if output is HIGH, false if LOW
+ */
+static inline bool ac_get_result(ac_registers_t *ac_reg, uint8_t channel_id)
+{
+	uint32_t state_mask = AC_STATUSA_STATE0_Msk << channel_id;
+
+	return (ac_reg->AC_STATUSA & state_mask) != 0;
+}
+
+/* Configure comparator channel: inputs, mode, hysteresis, output, interrupt, scaler, etc. */
+static int ac_configure_channel(const struct device *dev)
+{
+	/* Get device config */
+	const struct comparator_mchp_dev_config *const dev_cfg = DEV_CFG(dev);
```

Comment:
`DEV_CFG()` is not necessary, dev->config is a void *, so typecasting is implicit.